explainer.md

```diff
@@ -0,0 +1,447 @@
+      position: relative;
+      top: 100px;
+      width: 200px;
+      height: 200px;
+      background-color: green;
+      
+      border: 5px solid black;
+      border-radius: 50% 50%;
+      box-shadow: 0px 0px 5px;
+      
+      &:transition {
+        border: 5px solid transparent;
+        border-radius: none;
+        box-shadow: none;
+      }
+    }
+    
+    ::shared-container(#header-id) {
+        border: 5px solid black;
+        border-radius: 50% 50%;
+        box-shadow: 0px 0px 5px;
+    }
+    
+    ::shared-new(#header-id) {
+      top: -5px;
+      left: -5px;
+    }
+    
+    @keyframes border-animation {
+      from {
+        border: 10px solid black;
+        border-radius: 10% 10%;
+        box-shadow: 0px 0px 10px;
+      }
+    }
+    ::shared-container(#header-id) {
+      animation: ::shared-container-header-id 1s ease-in, border-animation 1s ease-in;
+    }
+    
+    @keyframes position-animation {
+      from {
+        top: -10px;
+        left: -10px;
+      }
+      to {
+        top: -5px;
+        left: -5px;
+      }
+    }
+    ::shared-old(#header-id), ::shared-new(#header-id) {
+      animation: position-animation 1s ease-in;
+    }
+    
+  </style>
+</head>
+<body>
+  <div class="animated" sharedId="header-id" id="header">Shared Element</div>
+</body>
+</html>
+```
+
+The additional steps taken by the browser in the example above are :
+
+i. When a pseudo element is created for shared elements in the old page in step 1, the pseudo-class "transition" is enabled for each shared element.
+
+ii. In step 3 when saving state from the old page, the completed ComputedStyle on the pseudo elements is saved in addition to the computed size and transform.
+
+iii. In step 7 when applying a UA stylesheet to pseudo elements on the new page, the saved style is also applied to the old pseudo elements :
+```
+::shared-old(#root) {
+  position: absolute;
+  width: 100%;
+  height: 100%;
+  
+  // This is the saved output referenced in step 3.
+  content: cached-element(html);
+  top: -10px;
+  left: -10px;
+}
+```
+
+## Modifications to element()
+The following changes will be made to the element() spec as a part of this proposal. The element captured by this function is the target element :
+
+* The target element must have paint containment (contain:paint) to ensure the element is the containing block for all positioned descendents and generates a stacking context.
+* The target element must disallow fragmentation (break-inside:avoid).
+* A new cached-element() function is introduced to refer to the saved output of the element() function in step 3) of the design.
+* Nested shared elements are omitted from the output of element() function.
+* The special cases when running the element() function on the html element are :
+    * The natural size for the generated image is the visual viewport bounds.
+    * When creating the image, the element is drawn on a canvas with the background color of the document.
```

Comment:
Also "shared elements captured via element() are not painted in the regular dom"

Reply:
Done.